lib/engine/route.rb

```diff
@@ -285,6 +299,13 @@ def connection_data
       @connection_data = []
       return @connection_data unless @connection_hexes
 
+      if @train.local? && @connection_hexes.one? && @connection_hexes[0].include?('local')
+        city_node = @game.hex_by_id(@connection_hexes[0][1]).tile.nodes.find do |n|
+          @game.city_tokened_by?(n, corporation)
+        end
+        return @connection_data << { left: city_node, right: city_node, connection: Connection.new }
```

Comment:
let's make a helper method for this, second time you're calling it

Reply:
Done.